rllib/offline/estimators/tests/test_ope.py

```diff
@@ -199,22 +217,440 @@ def test_fqe_model(self):
         tmp_probs = []
         for act in range(fqe.policy.action_space.n):
             tmp_actions = np.zeros_like(self.batch["actions"]) + act
-            log_probs = fqe.policy.compute_log_likelihoods(
+            log_probs = self.policy.compute_log_likelihoods(
                 actions=tmp_actions,
                 obs_batch=self.batch["obs"],
             )
-            tmp_probs.append(torch.exp(log_probs))
-        tmp_probs = torch.stack(tmp_probs).transpose(0, 1)
-        tmp_probs = convert_to_numpy(tmp_probs)
+            tmp_probs.append(np.exp(log_probs))
+        tmp_probs = np.stack(tmp_probs).T
         check(action_probs, tmp_probs, decimals=3)
 
-    def test_multiple_inputs(self):
-        # TODO (Rohan138): Test with multiple input files
-        pass
+    def test_fqe_optimal_convergence(self):
+        # Optimal CliffWalkingWallPolicy with epsilon = 0.0
+        # and CliffWalkingWallEnv are deterministic;
+        # check that FQE converges to the true Q-values for self.batch
+        q_vals = [
```

Comment:
How did you compute these q_values? do you wanna script that up here so it's clear to the next maintainer?

Reply:
**TODO**: what happened to this?

Reply:
I added the line above it-`        If self.batch["rewards"] = [-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 10],
        and gamma = 0.99, the discounted returns i.e. optimal Q-values are as follows:`

Reply:
Was this manually computed? I'm sure there was a script. Why don't you use the same script here?